var/spack/repos/builtin/packages/py-fiona/package.py

```diff
@@ -27,11 +28,13 @@ class PyFiona(PythonPackage):
     version("1.8.18", sha256="b732ece0ff8886a29c439723a3e1fc382718804bb057519d537a81308854967a")
 
     # pyproject.toml
-    depends_on("python@3.7:", when="@1.9:", type=("build", "link", "run"))
-    depends_on("python@2.6:", when="@1.8.22:1.8", type=("build", "link", "run"))
-    depends_on("python@2.6:3.10", when="@1.8.21", type=("build", "link", "run"))
-    depends_on("python@2.6:3.9", when="@:1.8.20", type=("build", "link", "run"))
-    depends_on("py-cython@0.29.29:0.29", when="@1.9:", type="build")
+    depends_on("python@:3.10", when="@1.8.21", type=("build", "link", "run"))
+    depends_on("python@:3.9", when="@:1.8.20", type=("build", "link", "run"))
+    depends_on("py-cython", type="build")
+    # Overly strict version requirements
+    # depends_on("py-cython@3.0.2:3", when="@1.9.5:", type="build")
+    # depends_on("py-cython@0.29.29:0.29", when="@1.9.0:1.9.4", type="build")
```

Comment:
@haampie not sure how you feel about this but your cython 3.0.2 PR isn't in yet so I didn't try to add the constraint. 1.9.5 builds with cython 0 and 1.9.4 builds with cython 3 so I think this can be relaxed.